library/checkout/afterpaycheckout.php

```diff
@@ -163,16 +163,16 @@ public function getShippingAddress()
 
             $phone = $this->getPhone($this->shipping_address);
 
-            if ($carrier->external_module_name == 'sendcloud') {
-                $sendCloudClassName = 'SendcloudServicePoint';
-                $service_point = $sendCloudClassName::getFromCart($this->cart->id);
-                $point = $service_point->getDetails();
-                $street = $point->street;
-                $houseNumber = $point->house_number;
-                $houseNumberSuffix = '';
-                $zipcode = $point->postal_code;
-                $city = $point->city;
-                $country = $point->country;
+            $carrierHandler = new CarrierHandler($this->cart);
+            $sendCloudData = $carrierHandler->handleSendCloud();
```

Comment:
why not create a $address array with the default shipping data and override the array if $sendCloudData exists, this will reduce the number of variables, and will make easier to read